pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -242,7 +245,63 @@ function subvolume_delete(volume, subvol, mount_point_in_parent, card) {
     });
 }
 
-export function make_btrfs_subvolume_page(parent, volume, subvol) {
+function dirname(path) {
+    const i = path.lastIndexOf("/");
+    if (i < 0)
+        return null;
+    else
+        return path.substr(0, i);
+}
+
+export function make_btrfs_subvolume_pages(parent, volume) {
+    let subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+    if (!subvols) {
+        const block = client.blocks[volume.path];
+        /*
+         * Try to show subvolumes based on fstab entries, this is a bit tricky
+         * as mounts where subvolid cannot be shown userfriendly.
```

Comment:
What about using `findmnt` for this task? For mounted subvolumes it shows:

```
[jelle@t14s][~/projects/cockpit/show-btrfs-tree]%findmnt --type btrfs
TARGET        SOURCE                         FSTYPE OPTIONS
/             /dev/mapper/system[/root]      btrfs  rw,relatime,compress=lzo,ssd,discard=async,space_cache,subvolid=256,subvol=/root
├─/.snapshots /dev/mapper/system[/snapshots] btrfs  rw,relatime,compress=lzo,ssd,discard=async,space_cache,subvolid=258,subvol=/snapshots
└─/home       /dev/mapper/system[/home]      btrfs  rw,relatime,compress=lzo,ssd,discard=async,space_cache,subvolid=257,subvol=/home
```

So the question is, is there a scenario where a subvolume is in fstab but it has children which we should show?

Reply:
Or even better, why don't we match `pathname` to whatever `btrfs subvolume list -ap` provides, that way we should figure out the subvolid and parent.

Reply:
But, nothing is mounted... I am confused.

Reply:
Ah, we still need to show an order when nothing is mounted, right. So what we do is then just fully arbitrary?

Reply:
Ah right I get it now, we basically show it based on pathname. That would be interesting with `subvol=root` on fedora.